### PR TITLE
Improve wolfram command

### PR DESCRIPTION
**Pull request by ontanj**
Resolves issue #998 
Populates result/data with unmodified :queryresult object from Wolfram API call.
Sends textual data to user when present, otherwise link to image.

### CODE REVIEW

File: src/yetibot/commands/wolfram.clj [hunk 1]
```diff
@@ -2,31 +2,63 @@
...
+  (let [plaintext (:plaintext subpod)]
+    (if-not (empty? plaintext)
+      plaintext
+      (get-in subpod [:img :src]))))
+
+(defn data-string
+  "Concatenate pod title and all subpods' data."
+  [pod]
+  (->> pod
+       (:subpods)
```
Comment (devth) [inline src/yetibot/commands/wolfram.clj:32]: This can simply be `:subpods`

File: src/yetibot/commands/wolfram.clj [hunk 2]
```diff
@@ -2,31 +2,63 @@
...
-         (parse-imgs-from-xml
-           (xml/parse (str endpoint "&input=" (encode q)))))))
+  (let [data (:queryresult (json/read-str
+                            (slurp (str endpoint "&input=" (encode q)))
+                            :key-fn keyword))]
+    (if (:success data)
+      {:result/value (map data-string (:pods data))
+       :result/collection-path [:pods]
+       :result/data data}
+      {:result/value (if (:error data)
```
Comment (devth) [inline src/yetibot/commands/wolfram.clj:57]: I think we would want to return `{:result/error (:error data)}` in this case to surface the error from wolfram, right?

File: src/yetibot/commands/wolfram.clj [hunk 3]
```diff
@@ -2,31 +2,63 @@
...
 (defn search-wolfram
   "wolfram <query> # search for <query> on Wolfram Alpha"
   {:yb/cat #{:info :img}}
   [{q :match}]
-  (flatten
-    (map #(str (second %) "&t=.jpg")
-         (parse-imgs-from-xml
-           (xml/parse (str endpoint "&input=" (encode q)))))))
+  (let [data (:queryresult (json/read-str
+                            (slurp (str endpoint "&input=" (encode q)))
```
Comment (devth) [inline src/yetibot/commands/wolfram.clj:51]: I realize the previous code was not great in terms of dynamically building a url. If you're looking to improve this, I'd probably switch to [clj-http](https://github.com/dakrone/clj-http) to clean up the http request. Something like:

```clojure
(client/get endpoint {:query-params {:input q :output "json" :appid app-id}
                      :as :json})
```

This handles query string encoding and json parsing of the response.

File: src/yetibot/commands/wolfram.clj (hunk 2 above)
Comment (ontanj) [reply src/yetibot/commands/wolfram.clj:57]: yes of course, I missed that option.

**Pull request merged**